File: models/Gpt-3.5-turbo/cpp/code/problems/advanced_techniques/Hamming_Distance.cpp

#include <iostream>
#include <vector>

using namespace std;

int hammingDistance(string a, string b) {
    int distance = 0;
    for (int i = 0; i < a.length(); i++) {
        if (a[i] != b[i]) {
            distance++;
        }
    }
    return distance;
}

int main() {
    int n, k;
    cin >> n >> k;
    
    vector<string> strings(n);
    for (int i = 0; i < n; i++) {
        cin >> strings[i];
    }
    
    int minDistance = k; // Initialize with maximum possible value
    
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int distance = hammingDistance(strings[i], strings[j]);
            minDistance = min(minDistance, distance);
        }
    }
    
    cout << minDistance << endl;
    
    return 0;
}

File: models/Gpt-3.5-turbo/py/code/advanced_techniques/Hamming_Distance.py

n, k = map(int, input().split())
strings = [input() for _ in range(n)]

min_distance = float('inf')

for i in range(n):
    for j in range(i+1, n):
        distance = bin(int(strings[i], 2) ^ int(strings[j], 2)).count('1')
        min_distance = min(min_distance, distance)

print(min_distance)
